emma-language/src/main/scala/org/emmalanguage/api/DataBag.scala

```diff
@@ -311,6 +316,34 @@ trait DataBag[A] extends Serializable {
           }
       })._1
   }
+
+  // -----------------------------------------------------
+  // equals, hashCode and toString
+  // -----------------------------------------------------
+
+  override def equals(o: Any) = o match {
+    case that: DataBag[A] => {
+      val thisVals = fetch()
+      val thatVals = that.fetch()
+      (thisVals.size == thatVals.size) &&
```

Comment:
At the moment, I think we would just use this probably for debugging, on small bags, so it's not really important to optimize it. But I'll add a TODO comment to make this smarter.

What we should actually do here in the long term is to write this without fetching at all. But this would complicate all the generated jobs that use this, and I don't want to do that at the moment.